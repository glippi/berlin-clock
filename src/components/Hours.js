import React from 'react'

export function Hours({ hours }) {
  const hourColor = hour => (hour === 'R' ? 'bg-red' : 'bg-silver')

  return (
    <div data-testid="hours">
      {hours.map((hour, i) => (
        <div key={i} className="flex flex-row justify-center items-center w5">
          {hour &&
            hour
              .split('')
              .map((h, idx) => (
                <div
                  key={idx}
                  className={`h3 ba pa2 ${hourColor(h)}`}
                  style={{ width: 'calc(100% / 4)' }}
                />
              ))}
        </div>
      ))}
    </div>
  )
}
