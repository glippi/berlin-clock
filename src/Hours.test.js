import React from 'react';
import { Hours } from './Hours';
import { render, cleanup } from '@testing-library/react';

afterEach(cleanup);

describe('Hours should', () => {
  test('render a red rectangle, three gray rectangles, a red rectangle and three gray rectangles', () => {
    const { container } = render(<Hours hours={["ROOO", "ROOO"]} />);

    expect(container.firstChild.firstChild).toMatchSnapshot();
  });
});
