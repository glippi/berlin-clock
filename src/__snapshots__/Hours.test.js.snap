// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hours should render a red rectangle, three gray rectangles, a red rectangle and three gray rectangles 1`] = `
<div
  class="flex flex-row justify-center items-center"
>
  <div
    class="w3 h3 ba pa2 bg-red"
  />
  <div
    class="w3 h3 ba pa2 bg-dark-gray"
  />
  <div
    class="w3 h3 ba pa2 bg-dark-gray"
  />
  <div
    class="w3 h3 ba pa2 bg-dark-gray"
  />
</div>
`;
